server/src/controllers/AuthController.js

```diff
@@ -151,28 +159,70 @@ module.exports = {
     }
   },
 
-  async verifyToken(req, res) {
+  async verifyToken(req, res, next) {
     const errors = validationResult(req);
 
     if (!errors.isEmpty()) {
       res.status(422).json({ errors: errors.array() });
       return;
     }
 
-    const { token } = req.body;
+    const token = req.headers.authorization.split(" ")[1];
 
     try {
-      jwt.verify(token, config.privateKey);
-      res.status(200).json({
-        status: true,
+      jwt.verify(token, config.privateKey, async (err, decoded) => {
+        if(!err){
+          next();
+        }else{
+          res.status(401).json({
+            status: false,
+            error: err.message
+          });
+        }
       });
     } catch (error) {
       res.status(403).json({
         status: false,
+        error: err.message
       });
     }
   },
 
+  async refreshAuthTokens(req, res){
+    try {
+      const { refreshToken, uid } = req.body;
+
+      if(!refreshToken || !uid){
+        return res.status(401).json({ error: "Data missing" });
+      }
+
+      const user = await User.findById(uid);
```

Comment:
This is a different object than what is initially stored as token. Maybe we can select the necessary fields and only use them throughout, both during initial creation, and during reissue.